packages/ui/src/components/DatePicker/SDatePicker.vue

```diff
@@ -0,0 +1,734 @@
+<script setup lang="ts">
+import MonthTable from './src/month-table.vue'
+import DatePanel from './date-panel.vue'
+import YearTable from './src/year-table.vue'
+import TimePanel from './time-panel.vue'
+
+import { IconBasicCheckMark24, IconArrowsChevronBottom24 } from '@/components/icons'
+// import { maska as vMaska } from 'maska'
+import { SPopover, SPopoverWrappedTransition } from '@/components/Popover'
+import { and } from '@vueuse/core'
+import { format, parse, isValid } from 'date-fns'
+import OptionsWrapper from './options-wrapper.vue'
+
+import { DatePickerType, RangeState, DateState, PickState, StateStore, ShowState, RangeOptionValue } from './types'
+
+interface Props {
+  modelValue: Date[] | Date
+  type: DatePickerType
+  time: boolean
+  disabled: boolean
+}
+const props = withDefaults(defineProps<Props>(), {
+  type: 'day',
+  time: false,
+  disabled: false,
+})
+
+const emit = defineEmits(['update:modelValue'])
+
+const isRange = computed(() => {
+  return props.type === 'range'
+})
+
+// #region STATE_STORE
+
+const today = new Date()
+
+const rangeState = ref<RangeState>({
+  selecting: false,
+  startDate: new Date(today.getFullYear(), today.getMonth(), 1) || null,
+  endDate: new Date(today.getFullYear(), today.getMonth(), 7) || null,
+})
```

Comment:
Take a look at [`toReactive()`](https://vueuse.org/shared/toreactive/). You may avoid some `.value` with it, but use actual `rangeStateRef` when you need to replace the object completely.